### DISCUSSION TITLE
Not all components are updated when property changes

### DISCUSSION THREAD

**Discussion by leirn** [slint-ui/slint#5996] [General]
Hi there,

I do observe a strange behaviour on my app.

The code below represents a simple part of the GUI that is used to change the property of an object.
```root.selected_layer``` is the currently selected value from a ```StandardItemList``` and is defined and updated this way : 

```
property <int> selected_layer;

layer_list := StandardListView {
    model: layers_list;
    current-item: root.selected_layer;
    current-item-changed(value) => {
        root.selected_layer = value;
    }
}
```

When I click on the list, the first two SpinBox (for "X:" and "Y:") are correctly updated. Unfortunately, the two sliders and the two other SpinBox are not, even if the code to update them six are absolutely the same...

```
Text {
    text: "Position";
}
HorizontalBox {
    Text {
        text: "X:";
    }
    SpinBox {
        width: 100px;
        maximum: map.width;
        value: root.layers[root.selected_layer].x / 1px;
        edited(value) => {
            root.layers[root.selected_layer].x = value * 1px;
        }
    }
}
Slider {
    minimum: 0;
    maximum: map.width;
    value: root.layers[root.selected_layer].x / 1px;
    changed(value) => {
        root.layers[root.selected_layer].x = value * 1px;
    }
}
HorizontalBox {
    Text {
        text: "Y:";
    }
    SpinBox {
        width: 100px;
        maximum: map.width;
        value: root.layers[root.selected_layer].y / 1px;
        edited(value) => {
            root.layers[root.selected_layer].y = value * 1px;
        }
    }
}
Slider {
    minimum: 0;
    maximum: map.height;
    value: root.layers[root.selected_layer].y / 1px;
    changed(value) => {
        root.layers[root.selected_layer].y = value * 1px;
    }
}
Text {
    text: "Transparency";
}
SpinBox {
    minimum: 0;
    maximum: 100;
    value: root.layers[root.selected_layer].transparency * 100;
    edited(value) => {
        root.layers[root.selected_layer].transparency = value / 100;
    }
}
Text {
    text: "Meters per pixel";
}
SpinBox {
    minimum: 1;
    maximum: 1000;
    value : root.layers[root.selected_layer].m_per_px;
    edited(value) => {
        root.layers[root.selected_layer].m_per_px = value;
    }
}
```

Any idea what could be the issue ?

Full code base can be seen here : 
https://github.com/leirn/mapannot.rs

Thanks

**Comment by hunger** [comment]
The UI looks nice (I just opened the URL of the main file taken from github in Slintpad) :-)

I could not spot the problematic elements in Slintpad though. Where in the UI are those? Can I reproduce the issue without the backend code?

**Reply by leirn** [reply]
These elements are in the file ui/appwindow.slint on ht egithub repo. They are located in the botom right of the application.

I have no idea if you can reproduce it without the backend since I am blind on the root cause of this effect

**Reply by Enyium** [reply]
This may be the same issue as #5990, and it would be quite important that this gets solved soon! I get the impression that `in-out` properties don't behave the way you'd expect them to.

@leirn, do you also work with backing properties without `<=>` two-way binding like I described in the issue? And does the issue depend on whether you manually control the widget with user input (so that the widget internally sets its property), or set a property of it that has a binding directly?

**Reply by leirn** [reply]
As I mentionned at the top of my message, the faulty property is defined as property <int> selected_layer;. It remains in the slint space as I do not need it in the rust part of the application.

It seems to act as if selected_layer update is not propagated to everywhere it should.

My knowledge of slint is too limited as I am a very recent user to try going further in the analysis.
